### PR TITLE
Kliot/admin menu

### PR DESCRIPTION
**Pull request by ekliot**
Added admin menu

Complete functionality:
* QuestMenu
* ItemMenu
* ArchetypeMenu
* RaceMenu

TODO
* Admin role granting

### CODE REVIEW

File: src/main/java/rpgcharacters/commands/CommandCreateTables.java [hunk 1]
```diff
@@ -237,22 +237,22 @@ private void seedRace( Connection conn ) {
...
+        archSeeds.put( "Warrior",   new int[]{  10,  10,   0,   5 } ); // 10 + 10 +  0 +  5 = 25
+        archSeeds.put( "Monk",      new int[]{   5,  10,   5,   5 } ); //  5 + 10 +  5 +  5 = 25
+        archSeeds.put( "Scoundrel", new int[]{   0,  15,   5,   5 } ); //  0 + 15 +  5 +  5 = 25
+        archSeeds.put( "Druid",     new int[]{  10,   5,  10,   0 } ); // 10 +  5 + 10 +  0 = 25
+        archSeeds.put( "Bard",      new int[]{   0,   5,  15,   5 } ); //  0 +  5 + 15 +  5 = 25
+        archSeeds.put( "Cleric",    new int[]{  10,   0,  15,   0 } ); // 10 +  0 + 15 +  0 = 25
+        archSeeds.put( "Sorceror",  new int[]{  20,   0,   5,   0 } ); // 20 +  0 +  5 +  0 = 25
+        archSeeds.put( "Magus",     new int[]{  15,  10,   0,   0 } ); // 15 + 10 +  0 +  0 = 25
+        archSeeds.put( "Shaman",    new int[]{  15,   0,   5,   5 } ); // 15 +  0 +  5 +  5 = 25
+        archSeeds.put( "Ranger",    new int[]{   0,  15,   0,  10 } ); //  0 + 15 +  0 + 10 = 25
```
Comment (returningsam) [inline src/main/java/rpgcharacters/commands/CommandCreateTables.java]: We may want to consider not seeding the db programatically and have an admin do it manually? Or not. that seems tedious, but just an idea.

File: src/main/java/rpgcharacters/userflow/AdminMenu.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                        Menu raceMenu = new RaceMenu( sc );
+                        raceMenu.enter( conn );
+                        break;
+                    case MENU_ARCH:
+                        Menu archetypeMenu = new ArchetypeMenu( sc );
+                        archetypeMenu.enter( conn );
+                        break;
+                    case MENU_GRANT:
+                        // Menu grantAdminMenu = new GrantAdminMenu( sc );
+                        // grantAdminMenu.enter( conn );
```
Comment (returningsam) [inline src/main/java/rpgcharacters/userflow/AdminMenu.java]: Im assuming these are commented out till role integration is complete?

Comment (ekliot) [inline src/main/java/rpgcharacters/userflow/AdminMenu.java]: affirmative

File: src/main/java/rpgcharacters/commands/CommandCreateTables.java (hunk 1 above)
Comment (ekliot) [inline src/main/java/rpgcharacters/commands/CommandCreateTables.java]: I figured these are the "baseline" archetypes and races for the arbitrary RPG system, so they would always exist regardless of the admin

File: src/main/java/rpgcharacters/userflow/ArchetypeMenu.java [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+                proficiency_mod = sc.nextInt();
+                loop = false;
+            } catch ( InputMismatchException e ) {
+                System.out.println( "Invalid input..." );
+                // scrub the Scanner
+                sc.nextLine();
+            }
+        } while ( loop );
+
+        // take input for power modifier
```
Comment (mwitte98) [inline src/main/java/rpgcharacters/userflow/ArchetypeMenu.java]: personality

File: src/main/java/rpgcharacters/userflow/ArchetypeMenu.java [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+                personality_mod = sc.nextInt();
+                loop = false;
+            } catch ( InputMismatchException e ) {
+                System.out.println( "Invalid input..." );
+                // scrub the Scanner
+                sc.nextLine();
+            }
+        } while ( loop );
+
+        // take input for power modifier
```
Comment (mwitte98) [inline src/main/java/rpgcharacters/userflow/ArchetypeMenu.java]: perception

File: src/main/java/rpgcharacters/userflow/ArchetypeMenu.java [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+            printArchetype( name, new int[]{
+                power_mod, proficiency_mod, personality_mod, perception_mod
+            } );
+            success = true;
+        } catch ( SQLException e ) {
+            e.printStackTrace();
+        }
+
+        if ( !success ) {
+            System.out.println( "Could not create archetype, " + name + "\n" );
```
Comment (mwitte98) [inline src/main/java/rpgcharacters/userflow/ArchetypeMenu.java]: Couldn't this be moved into the `catch` a couple lines above?

File: src/main/java/rpgcharacters/userflow/ItemMenu.java [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+            System.out.println( "" );
+        } catch ( SQLException e ) {
+            e.printStackTrace();
+        }
+    }
+
+    private void printItem ( String name, String quest, String desc ) {
+        System.out.println( "Name: " + name );
+        // TODO this line is being finicky
+        // System.out.println( "Rewarded for: " + ( quest.equals( "null" ) ? "Not rewarded" : quest ) );
```
Comment (mwitte98) [inline src/main/java/rpgcharacters/userflow/ItemMenu.java]: Have you tried `quest == null` with no quotes around null?

File: src/main/java/rpgcharacters/userflow/ItemMenu.java [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+            stmt.execute( query );
+
+            System.out.println( "Item " + name + " has been created!\n" );
+            success = true;
+        } catch ( SQLException e ) {
+            e.printStackTrace();
+        }
+
+        if ( !success ) {
+            System.out.println( "Could not create item, " + name + "\n" );
```
Comment (mwitte98) [inline src/main/java/rpgcharacters/userflow/ItemMenu.java]: Same with success

File: src/main/java/rpgcharacters/userflow/QuestMenu.java [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+            stmt.execute( query );
+
+            System.out.println( "Quest \"" + name + "\" has been created!\n" );
+            success = true;
+        } catch ( SQLException e ) {
+            e.printStackTrace();
+        }
+
+        if ( !success ) {
+            System.out.println( "Could not create quest, \"" + name + "\"\n" );
```
Comment (mwitte98) [inline src/main/java/rpgcharacters/userflow/QuestMenu.java]: Same with success

File: src/main/java/rpgcharacters/userflow/QuestMenu.java [hunk 9]
```diff
@@ -0,0 +1,379 @@
...
+            Statement stmt = conn.createStatement();
+            stmt.execute( query );
+
+            System.out.println( "Item " + name + " has been created!\n" );
+            success = true;
+        } catch ( SQLException e ) {
+            e.printStackTrace();
+        }
+
+        if ( !success ) {
```
Comment (mwitte98) [inline src/main/java/rpgcharacters/userflow/QuestMenu.java]: Same with success

File: src/main/java/rpgcharacters/userflow/RaceMenu.java [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+                proficiency_mod = sc.nextInt();
+                loop = false;
+            } catch ( InputMismatchException e ) {
+                System.out.println( "Invalid input..." );
+                // scrub the Scanner
+                sc.nextLine();
+            }
+        } while ( loop );
+
+        // take input for power modifier
```
Comment (mwitte98) [inline src/main/java/rpgcharacters/userflow/RaceMenu.java]: personality

File: src/main/java/rpgcharacters/userflow/RaceMenu.java [hunk 11]
```diff
@@ -0,0 +1,267 @@
...
+                personality_mod = sc.nextInt();
+                loop = false;
+            } catch ( InputMismatchException e ) {
+                System.out.println( "Invalid input..." );
+                // scrub the Scanner
+                sc.nextLine();
+            }
+        } while ( loop );
+
+        // take input for power modifier
```
Comment (mwitte98) [inline src/main/java/rpgcharacters/userflow/RaceMenu.java]: perception

File: src/main/java/rpgcharacters/userflow/RaceMenu.java [hunk 12]
```diff
@@ -0,0 +1,267 @@
...
+            System.out.println( "Race has been created!\n" );
+            printRace( name, new int[]{
+                power_mod, proficiency_mod, personality_mod, perception_mod
+            } );
+            success = true;
+        } catch ( SQLException e ) {
+            e.printStackTrace();
+        }
+
+        if ( !success ) {
```
Comment (mwitte98) [inline src/main/java/rpgcharacters/userflow/RaceMenu.java]: Same with success

File: src/main/java/rpgcharacters/userflow/ArchetypeMenu.java [hunk 13]
```diff
@@ -0,0 +1,267 @@
...
+    public ArchetypeMenu( Scanner sc ) {
+        this.sc = sc;
+        this.options = Arrays.asList( ARCH_LIST, ARCH_NEW, EXIT );
+    }
+
+    private void listArchetype() {
+        try {
+            // get every archetype, and the name of the quest that rewards it
+            String query = "SELECT *"
+                         + "FROM archetype";
```
Comment (mwitte98) [inline src/main/java/rpgcharacters/userflow/ArchetypeMenu.java]: Need a space between `*` and `FROM`. Don't really need to have it on two separate lines

File: src/main/java/rpgcharacters/userflow/RaceMenu.java [hunk 14]
```diff
@@ -0,0 +1,267 @@
...
+    public RaceMenu( Scanner sc ) {
+        this.sc = sc;
+        this.options = Arrays.asList( RACE_LIST, RACE_NEW, EXIT );
+    }
+
+    private void listRaces() {
+        try {
+            // get every race, and the name of the quest that rewards it
+            String query = "SELECT *"
+                         + "FROM race";
```
Comment (mwitte98) [inline src/main/java/rpgcharacters/userflow/RaceMenu.java]: Need a space between `*` and `FROM`

File: src/main/java/rpgcharacters/userflow/ItemMenu.java (hunk 6 above)
Comment (ekliot) [inline src/main/java/rpgcharacters/userflow/ItemMenu.java]: this works

File: src/main/java/rpgcharacters/userflow/RaceMenu.java (hunk 14 above)
Comment (ekliot) [inline src/main/java/rpgcharacters/userflow/RaceMenu.java]: oddly enough, this was working fine, but it is fixed

**Pull request merged**